typescript/infra/src/utils/gcloud.ts

```diff
@@ -95,6 +101,53 @@ export async function gcpSecretExists(secretName: string) {
   return matches.length > 0;
 }
 
+/**
+ * Uses the SecretManagerServiceClient to check if a secret exists.
+ * @param secretName The name of the secret to check.
+ * @returns A boolean indicating whether the secret exists.
+ */
+export async function gcpSecretExistsUsingClient(
+  secretName: string,
+  client?: SecretManagerServiceClient,
+): Promise<boolean> {
+  if (!client) {
+    client = new SecretManagerServiceClient({
+      projectId: GCP_PROJECT_ID,
+    });
+  }
+
+  try {
+    const fullSecretName = `projects/${await getCurrentProjectNumber()}/secrets/${secretName}`;
+    const [secrets] = await client.listSecrets({
+      parent: `projects/${GCP_PROJECT_ID}`,
+      filter: `name=${fullSecretName}`,
+    });
+
+    return secrets.length > 0;
+  } catch (e) {
+    console.error(`Failed to list secrets: ${e}`);
```

Comment:
think we've tended to try to use pino logging in these `src` files (see debugLog) - maybe you can use pino for the error log

Reply:
Is this what you mean?